corehq/apps/locations/permissions.py

```diff
@@ -20,34 +21,57 @@ def is_locations_admin(view_fn):
     return locations_access_required(domain_admin_required(view_fn))
 
 
-def user_can_edit_location(user, location, project):
-    """
-    Expects SQLLocation
-    """
-    if not project.location_restriction_for_users:
+@quickcache(['user._id', 'project.name'])
+def editable_locations(user, project):
+    if (user.is_domain_admin(project.name) or
+            not project.location_restriction_for_users):
+        return [l._id for l in Location.by_domain(project.name)]
+
+    user_loc = user.get_location(project.name)
+    if not user_loc:
+        return []
+
+    return [l.location_id for l in
+            user_loc.sql_location.get_descendants(include_self=True)]
```

Comment:
Good points. I almost forget about couch vs sql being in vellum most of the time